src/modules/iotjs_module_spi.c

```diff
@@ -342,18 +342,19 @@ JHANDLER_FUNCTION(SpiConstructor) {
 
 // FIXME: do not need transferArray if array buffer is implemented.
 JHANDLER_FUNCTION(TransferArray) {
-  DJHANDLER_CHECK_THIS(object);
+  JHANDLER_DECLARE_THIS_PTR(spi, spi);
+
   DJHANDLER_CHECK_ARGS(2, array, array);
   DJHANDLER_CHECK_ARG_IF_EXIST(2, function);
 
   const iotjs_jval_t* jcallback = JHANDLER_GET_ARG_IF_EXIST(2, function);
-  const iotjs_jval_t* jspi = JHANDLER_GET_THIS(object);
-  iotjs_spi_t* spi = iotjs_spi_get_instance(jspi);
 
   iotjs_spi_set_array_buffer(spi, JHANDLER_GET_ARG(0, array),
                              JHANDLER_GET_ARG(1, array));
 
   if (jcallback) {
+    const iotjs_jval_t* jspi = JHANDLER_GET_THIS(object);
+
     SPI_ASYNC(transfer, jspi, jcallback, kSpiOpTransfer);
```

Comment:
I'd modify `iotjs_spi_reqwrap_create` to accept the `iotjs_spi_t*`. It will be used that way later.